test/include/seqan3/test/tmp_filename.hpp

```diff
@@ -23,6 +23,23 @@
 
 #include <seqan3/std/filesystem>
 
+#if _WIN32
+
+#   include <stdio.h>
+    char *mkdtemp(char *template_name)
+    {
+        // return zero on success; an error code on failure.
+        if (_mktemp_s(template_name, strlen(template_name)+1))
+            return  nullptr;
+
+        // upss... hope no one creates my dir now
+        if (std::filesystem::create_directories(template_name))
+                return template_name ;
+
+        return  nullptr;
+    }
+#endif
+
 namespace seqan3
 {
 namespace test
```

Comment:
Add the following here such that it looks like this:

```cpp
namespace seqan3
{
namespace test
{
#if defined(_WIN32)
char * mkdtemp(char * template_name)
{
	if (_mktemp_s(template_name, strlen(template_name) + 1))
		return nullptr;
	
	if (std::filesystem::create_directories(template_name))
		return template_name;
		
	return nullptr;
}
#endif
```

Reply:
Interesting. Are you sure you want mkdtemp to be an internal function in windows but global in others OS? This may led to differences in how to write tests for different OS.  Just confirm this.

Reply:
The name lookup will first look in the current namespace, then the enclosing one, and so on.
Putting something in the global namespace may have consequences, e.g. when someone else defines mkdtemp on a global level.

Reply:
Even is this "something" is implementing things expected to be originally in the global namespace? Maybe OK - your decision. 
> may have consequences, e.g. when someone else defines mkdtemp on a global level.

And it will be great if the consequence will be a compiler error or warning - for example if MS/minwg/gcc decide to include a mkdtemp in the stdlib for windows. Otherwise we will continue to use our workaround unnoticed.

Reply:
In the context we use it, we don't need it in the global namespace.

> And it will be great if the consequence will be a compiler error or warning - for example if MS/minwg/gcc decide to include a mkdtemp in the stdlib for windows. Otherwise we will continue to use our workaround unnoticed.

And it would force everyone who has a `mkdtemp` in the global namespace to either alter our code our remove their implementation.

Since we only use `mkdtemp` in this one function, exactly only in our test header, it would IMO be bad style to do provide this function globally.

Reply:
OK, changed. 
Just for the record: what if some other stdlib in windows provide `mkdtemp `or `std::aligned_alloc` or... we will need to be more specific than just `#if defined(_WIN32)`

Reply:
For the tests, we will always use our `mkdtemp` since it is found first (we're calling `mkdtemp` and not `::mkdtemp`) - so this will be fine.

`std::aligned_alloc` may be reevaluated when it's actually available...